### ISSUE TITLE
View.cornerRadius prevents introspectScrollView from working

### ISSUE THREAD

**Issue by joshuakcockrell** [siteline/SwiftUI-Introspect#139]
Introspect 0.1.3
Xcode 13.2.1

Very similar to https://github.com/siteline/SwiftUI-Introspect/pull/83 and https://github.com/siteline/SwiftUI-Introspect/issues/93

but I wanted to provide another example (`ScrollView` in this case though)
Feel free to mark as duplicate!

This example has 2 nested `ScrollView`s. But if the `.cornerRadius` is present, both `introspectScrollView` calls result in the same exact `ScrollView`. Remove the `.cornerRadius` and everything works fine.

    struct ContentView: View {
        
        var body: some View {
            
            ScrollView {
                Text("outer ScrollView")
                
                ScrollView {
                    Text("inner ScrollView")
                }
                    .cornerRadius(20) // Comment / uncomment this to see breaking behavior
                    .introspectScrollView { scrollView in
                        print("Inner addr: \(String(format: "%p", scrollView))")
                    }
            }
            .introspectScrollView { scrollView in
                print("Outer addr: \(String(format: "%p", scrollView))")
            }
        }
    }

You know they are the same `ScrollView` from the prints:

    Outer addr: 0x7f848902e200
    Inner addr: 0x7f848902e200

If you remove `.cornerRadius` you get different pointers:

    Outer addr: 0x7fcbb5830000
    Inner addr: 0x7fcbb5812800

**Comment by joshuakcockrell** [comment]
It seems I was able to fix this by nesting the `.introspectScrollView` inside the `ScrollView`. 

    ScrollView {                                
        Color.clear.frame(width: 0, height: 0)
            .introspectScrollView { scrollView in

Not sure how fragile this is though cause it's probably relying on the search-up-the-ancestor-tree fallback right?

**Comment by ostatnicky** [comment]
I've got the same issue, the `.introspectScrollView` is not called with `.cornerRadius`.

Luckily, I found out that `.background` with `RoundedRectangle` work without any problem with the same UI result 💪 

```
ScrollView {
}
.introspectScrollView { scrollView in
    scrollView.alwaysBounceVertical = false
}
.background(
    RoundedRectangle(cornerRadius: 16)
        .fill(Color.yellow)
    )
}
```

**Comment by paescebu** [comment]
should be fixed with 0.3.1 :)

**Comment by davdroman** [comment]
@joshuakcockrell as you may now, this is now a thing of the past when using the new [SwiftUIIntrospect](https://github.com/siteline/SwiftUI-Introspect#readme) module:

```swift
struct ContentView: View {
    var body: some View {
        ScrollView {
            Text("outer ScrollView")

            ScrollView {
                Text("inner ScrollView")
            }
            .cornerRadius(20) // Comment / uncomment this to see fixed behavior
            .introspect(.scrollView, on: .iOS(.v13, .v14, .v15, .v16, .v17)) { scrollView in
                print("Inner addr: \(String(format: "%p", scrollView))")
            }
        }
        .introspect(.scrollView, on: .iOS(.v13, .v14, .v15, .v16, .v17)) { scrollView in
            print("Outer addr: \(String(format: "%p", scrollView))")
        }
    }
}
```

Result:

```
Inner addr: 0x124833200
Outer addr: 0x124818c00
Inner addr: 0x124833200
Outer addr: 0x124818c00
```

Closing for now :)

##### Note: the print repetition is due to the fact the customize block isn't guaranteed to be called any specific number of times (e.g. just once), so you should account for that when implementing your customization logic to not repeat CPU or memory intensive operations needlessly.
